src/ProgressOnderwijsUtils/Data/ParameterizedSqlObjectMapper.cs

```diff
@@ -78,8 +79,13 @@ public static int ExecuteNonQuery(this ParameterizedSql sql, [NotNull] SqlComman
                 SqlDataReader reader = null;
                 try {
                     reader = cmd.Command.ExecuteReader(CommandBehavior.SequentialAccess);
-                    var unpacker = DataReaderSpecialization<SqlDataReader>.ByMetaObjectImpl<T>.DataReaderToRowArrayUnpacker(reader, fieldMapping);
-                    return unpacker(reader, out lastColumnRead);
+                    var unpacker = DataReaderSpecialization<SqlDataReader>.ByMetaObjectImpl<T>.DataReaderToSingleRowUnpacker(reader, fieldMapping);
+                    var builder = new ArrayBuilder<T>();
+                    while (reader.Read()) {
+                        var nextRow = unpacker(reader, out lastColumnRead);
+                        builder.Add(nextRow);
+                    }
+                    return builder.ToArray();

```

Comment:
This is kind of the heart of this PR: instead of compiling a delegate that loops to create an array, compile a delegate that creates a row, and call that in a loop.